debugger/show_source.ml

```diff
@@ -23,18 +23,23 @@ open Source
 
 (* Print a line; return the beginning of the next line *)
 let print_line buffer line_number start point before =
-  let next = next_linefeed buffer start
+  let next_linefeed = next_linefeed buffer start
   and content = buffer_content buffer
   in
     printf "%i " line_number;
+    let next =
+      if content.[next_linefeed - 1] = '\r' then
```

Comment:
I thought I'd persuaded that couldn't happen (that the function wouldn't be called in that context) but even if that's right, it would be better to be defensive.